### PR TITLE
Add ability to download and open files

### PR DESCRIPTION
**Pull request by paul-nameless**
- introduced new entity MsgProxy: it adds helpful fields to make writing less code
- extended lib with new api call download_file
- rewrote msg handling
- rewrote msg parsing, to have one function for chats and msgs
- open msg with `l` (open should be installed, mailcap support in next patches)
- added update_file handler for updating file download state
- removed tdlib logging to stdin

### CODE REVIEW

File: tg/controllers/__init__.py [hunk 1]
```diff
@@ -25,20 +21,54 @@ class Controller:
...
+                    s.call(["less", f.name])
+            return
+
+        path = msg.local_path
+        if path:
+            # handle with mimetype and mailcap
+            # if multiple entries in mailcap, open fzf to choose
+            with suspend(self.view) as s:
+                log.info("Opening file: %s", path)
+                s.call(["open", path])
```
Comment (lightme16) [inline tg/controllers/__init__.py:70]: this works only for mac os, we need to use `xdg-open` when on linux 
https://stackoverflow.com/a/264401/5187245

**Comment by paul-nameless** [comment]
For this patch to work, there should be mailcap file at `~/.mailcap`:
```
#### images
image/png; [file_handler_program] "%s"
image/jpeg; [file_handler_program] "%s"
image/pjpeg; [file_handler_program] "%s"
image/jpg; [file_handler_program] "%s"
image/gif; [file_handler_program] "%s"

#### media
audio/ogg; mpv "%s"
video/mp4; mpv "%s"
audio/mpeg; mpv --no-video "%s"

#### text
text/html; w3m "%s"
text/plain; less "%s"

# fallback to vim
text/*; vim "%s"
```

And config file at `~/.config/tg/tg.conf`:
```
[DEFAULT]
api_id = [id]
api_hash = [hash]
login = [phonne]
enc_key = [encrytpion_key]
notify_cmd = /usr/local/bin/terminal-notifier -title "{title}" -subtitle "{subtitle}" -message "{msg}" -appIcon "{icon_path}" -sound default
```

File: tg/controllers/__init__.py [hunk 2]
```diff
@@ -163,41 +196,68 @@ def refresh_msgs(self) -> None:
...
-        # message_text = message_content.get('text', '').lower()
-
-        # if message_text == 'ping':
-        #     chat_id = update['message']['chat_id']
-        #     # print(f'Ping has been received from {chat_id}')
-        #     self.tg.send_message(
-        #         chat_id=chat_id,
-        #         text='pong',
-        #     )
+                if proxy.is_downloaded():
```
Comment (lightme16) [inline tg/controllers/__init__.py:261]: what's why we need mypy
> TypeError: 'bool' object is not callable

File: tg/main.py [hunk 3]
```diff
@@ -1,63 +1,65 @@
...
     log.debug("#" * 64)
-    tg = Telegram(
-        api_id=API_ID,
-        api_hash=API_HASH,
-        phone=PHONE,
-        database_encryption_key="changeme1234",
+    tg = TelegramApi(
+        api_id=cfg["api_id"],
+        api_hash=cfg["api_hash"],
+        phone=cfg["login"],
```
Comment (lightme16) [inline tg/main.py:56]: login is confusing, let's use `phone` as a config name

**Pull request merged**